source/features/one-key-formatting.tsx

```diff
@@ -7,24 +7,32 @@ import {onCommentFieldKeydown, onConversationTitleFieldKeydown, onCommitTitleFie
 
 const formattingCharacters = ['`', '\'', '"', '[', '(', '{', '*', '_', '~', '“', '‘'];
 const matchingCharacters = ['`', '\'', '"', ']', ')', '}', '*', '_', '~', '”', '’'];
+const quoteCharacters = new Set(['`', '\'', '"']);
 
 function eventHandler(event: DelegateEvent<KeyboardEvent, HTMLTextAreaElement | HTMLInputElement>): void {
 	const field = event.delegateTarget;
+	const formattingChar = event.key;
 
-	if (!formattingCharacters.includes(event.key)) {
+	if (!formattingCharacters.includes(formattingChar)) {
 		return;
 	}
 
 	const [start, end] = [field.selectionStart, field.selectionEnd];
 
 	// If `start` and `end` of selection are the same, then no text is selected
-	if (start === end) {
+	if (start === end || start === null || end === null) {
+		return;
+	}
+
+	const selectedText = field.value.slice(start, end);
+
+	// Allow replacing quotes
+	if (quoteCharacters.has(formattingChar) && quoteCharacters.has(selectedText)) {
```

Comment:
Since `.slice` can be a heavy operation on long fields, you can avoid running it altogether in most cases by using this:

```suggestion
	if (quoteCharacters.has(formattingChar) && quoteCharacters.has(textFieldEdit.getSelection(field))) {
```

But in reality it's even cheaper to use a single character check via `end - start === 1 && field.value.at(start)` because this should only work on a single character anyway.

Reply:
Not too readable, but if it's better for performance, let's use that.